### PR TITLE
Implement trapezoid function in keras.ops

### PR DESCRIPTION
**Pull request by shashaka**
Adds keras.ops.trapezoid, which performs numerical integration using the trapezoidal rule along a specified axis.
Supported across NumPy, TensorFlow, PyTorch, and JAX backends.
Not supported on OpenVINO.

### CODE REVIEW

File: keras/src/ops/numpy_test.py [hunk 1]
```diff
@@ -3604,6 +3615,21 @@ def test_any(self):
...
+
+        self.assertAllClose(knp.trapezoid(y), np.trapezoid(y))
+        self.assertAllClose(knp.trapezoid(y, x), np.trapezoid(y, x))
+        self.assertAllClose(
+            knp.trapezoid(y, x, dx=dx), np.trapezoid(y, x, dx=dx)
+        )
+        self.assertAllClose(
+            knp.trapezoid(y, x, dx=dx, axis=1),
+            np.trapezoid(y, x, dx=dx, axis=1),
+        )
```
[the remark this answers is not in the document]
Comment (shashaka) [reply keras/src/ops/numpy_test.py:3629]: Updated

File: keras/src/backend/tensorflow/numpy.py [hunk 2]
```diff
@@ -2984,6 +2984,34 @@ def transpose(x, axes=None):
...
 
 
+def trapezoid(y, x=None, dx=1.0, axis=-1):
+    def _move_axis_to_last(tensor, axis):
+        if axis != -1 and tensor.shape.rank is not None:
+            perm = list(range(tensor.shape.rank))
+            perm.pop(axis)
+            perm.append(axis)
+            tensor = tf.transpose(tensor, perm)
+        return tensor
```
[the remark this answers is not in the document]
Comment (shashaka) [reply keras/src/backend/tensorflow/numpy.py]: Updated

File: keras/src/ops/numpy.py [hunk 3]
```diff
@@ -7088,6 +7088,43 @@ def transpose(x, axes=None):
...
+    Args:
+        y: Input tensor.
+        x: Optional tensor specifying sample points corresponding to `y`.
+           If `None`, spacing is assumed to be `dx`.
+        dx: Spacing between sample points when `x` is `None`.
+        axis: Axis along which to integrate. Default is the last axis.
+
+    Returns:
+        The approximate integral of `y` along the given axis.
+    """
```
[the remark this answers is not in the document]
Comment (shashaka) [reply keras/src/ops/numpy.py:7127]: Updated
